src/poetry/utils/env/base_env.py

```diff
@@ -312,7 +349,9 @@ def run_pip(self, *args: str, **kwargs: Any) -> str:
     def run_python_script(self, content: str, **kwargs: Any) -> str:
         return self.run(
             self._executable,
-            "-I",
+            # if isolation is enabled for system env, user site would be
```

Comment:
I suppose this reintroduces #6627, which was the reason for adding this flag

Reply:
Good catch thanks. Let me see if there is a better way.